File: ZMQ(recommended)/octave/publisher.m
% --------------- Looad libraries ----------
 base_path = "C:/nep-matlab";
 javaaddpath ([base_path, "/nep.jar"]);
 javaaddpath ([base_path, "/jeromq-0.4.3.jar"]);
 javaaddpath ([base_path, "/gson-2.8.4.jar"]);
 addpath('jsonlab');
 
% ---------------- New node ----------------
node = javaObject ("nep.Node", "octave_node")

% ------------ Define subscriber -----------
pub = node.new_pub("test", "json");


for c = 1:5
   msg = struct('message','hello','value',c) %Messages to send as structures (python-like dictionaries)
   json_msg = savejson(msg)
   pub.publish(json_msg) 
   pause(1)  
end

% ---------------- Close publisher -----------------
pub.close()  % the subscriber must to be closed.

File: ZMQ(recommended)/python/publish-subscribe/master_local/publisher.py
import nep
import time
import sys

msg_type = "json"         # Message type to listen. "string" or "json"

node = nep.node("publisher_sample")             # Create a new node
pub = node.new_pub("test",msg_type)        # Set the topic and the configuration of the publisher

# Publish a message each second
i = 0
while True: 
    #  --- String example ---
    if msg_type == "string":
        msg = "hello world"
        print ("sending: " + msg)
        pub.publish(msg) 
        time.sleep(1)
    # --- JSON example ---
    if msg_type == "json":
        i = i + 1
        msg =  data = {"message":"hello"}
        print ("sending: " + str(msg))
        pub.publish(msg) 
        time.sleep(1)
